comid/comid.go

```diff
@@ -252,26 +272,50 @@ func (o Comid) ToCBOR() ([]byte, error) {
 		return nil, err
 	}
 
-	return encoding.SerializeStructToCBOR(em, &o)
+	var entities *Entities
+	if o.Entities != nil && o.Entities.IsEmpty() {
```

Comment:
The whole block is extremely confusing here. Suggest renaming to some better naming to clarify the concept!

Reply:
Also I am not clear as to why this is not an error? Essentially Entire Collection object is empty however a pointer is set ?

Reply:
`Entities` is a `Collection`, as well as containing elements of some type, it also caches extensions for those elements. If extensions have been registered, but no elements have been added, the collection will be empty but it will still exist to keep track of the extensions (the pointer won't be `nil`). 

Since we want to omit the claim, rather than marshaling an empty list, we temporarily set the field to `nil`, let the normal marhaling take place, and then restore it afterwards.

I can rename to local vairable to something like `temp` or `save` but I'm not really sure that would make things clearer?

Reply:
Oh! actually, since teh receiver is value and not pointer, we don't actually need to save a local copy. We can just set it to nil. Since the `o` that exists in this method was passed by value and is a copy of the caller, modifying the former won't affect the latter.